baselines/dom.generated.d.ts

```diff
@@ -1485,9 +1485,18 @@ interface RTCTransportStats extends RTCStats {
     tlsVersion?: string;
 }
 
-interface ReadableStreamReadDoneResult {
+interface ReadableStreamGetReaderOptions {
+    /**
+     * Creates a ReadableStreamBYOBReader and locks the stream to the new reader.
+     *
+     * This call behaves the same way as the no-argument variant, except that it only works on readable byte streams, i.e. streams which were constructed specifically with the ability to handle "bring your own buffer" reading. The returned BYOB reader provides the ability to directly read individual chunks from the stream via its read() method, into developer-supplied buffers, allowing more precise control over allocation.
+     */
+    mode?: ReadableStreamReaderMode;
+}
+
+interface ReadableStreamReadDoneResult<T> {
     done: true;
-    value?: undefined;
+    value?: T;
 }
```

Comment:
This is needed for BYOB readers, which return an empty view on the original buffer when the stream is closed.

I'm not sure if we still want separate types for `done: false` and `done: true`? On the one hand, it's nice that you can write:
```javascript
while (true) {
  const result = await reader.read();
  if (result.done) {
    break;
  }
  const chunk = result.value; // narrowed to T, cannot be undefined
}
```
On the other hand, I don't know if this a sufficiently convincing argument to keep this in TypeScript's DOM types. The stream specification itself only has [`ReadableStreamReadResult`](https://streams.spec.whatwg.org/#dictdef-readablestreamreadresult), which does not distinguish between the two cases.